Gifski/AppDelegate.swift

```diff
@@ -125,7 +125,13 @@ extension NSColor {
 @NSApplicationMain
 final class AppDelegate: NSObject, NSApplicationDelegate {
 	@IBOutlet private weak var window: NSWindow!
-	let gifski = Gifski()
+	var gifski: Gifski? {
```

Comment:
Does `gifski` really need to be an instance variable now? Couldn't we just initiate it in the `startConversion` function and have the `onProgress` callback there too?

Like:

```swift
…

progress = Progress(totalUnitCount: 1)
progress?.becomeCurrent(withPendingUnitCount: 1)
let gifski = Gifski.convertFile(
	inputUrl,
	outputUrl: outputUrl,
	quality: defaults["outputQuality"] as! Double,
	dimensions: choosenDimensions,
	frameRate: choosenFrameRate
)
gifski.onProgress = { [weak self] progress in
	self?.updateProgress(progress)
}
progress?.resignCurrent()

…
```

Reply:
True. I dont think progress needs to either..

I kind of want to Gifski.convertFile to just be a function too.. Something like:

`static convert(fileAt:outputTo:withQuality:dimensions:frameRate:completionHandler:)`

I have a proof of concept I can submit to explain better..